eeprom/eeprom.c

```diff
@@ -56,12 +56,12 @@ void write_log_entry(const char *str, uint8_t *index) {
         printf("Maximum log entries. Erasing the log to log the messages\n");
         erase_logs(index);
     }
-    size_t size_length = strlen(str); //not include NULL terminator
+    size_t size_length = strlen(str) + 1; //include NULL terminator
 
-    if (size_length > STRLEN_EEPROM - 1) {
-        size_length = STRLEN_EEPROM - 1;
+    if (size_length > STRLEN_EEPROM) {
+        size_length = STRLEN_EEPROM;
     }
-    uint8_t log_buf[size_length + 3];
+    uint8_t log_buf[size_length + 2];
```

Comment:
+2 has a macro ADDR_LEN

Reply:
Also.. size_length? I'm assuming string_length

Reply:
that +2 is 2 characters of crc16 so it's nothing related to eeprom ADDR_LEN in macro